### PR TITLE
Add a method to retrieve scores on delegates and sort them on a list

### LINKED ISSUE #659: Add top delegates on delegate page
On the delegate page for space here: https://snapshot.org/#/delegate/balancer.eth

We should show the top delegate in a new section like this:
image

Delegations can be taken from here: https://thegraph.com/legacy-explorer/subgraph/snapshot-labs/snapshot
We should

 1: Take relations delegator / delegate
 2: Calculate delegator voting power using getScores and current space strategies
 3: Sort delegates by voting power

This section should not be visible on that page: https://snapshot.org/#/delegate

### PR DESCRIPTION
**Pull request by gbonumore**
Fixes #659

Changes proposed in this pull request:
Added a method on Delegate to add scores on delegates and sort them by score, if they have any.
Since I am missing in balance on testnet on spaces with delegation enabled I am marking this as WIP.

### CODE REVIEW

**Comment by samuveth** [comment]
Hi @gbonumore this doesn't seem to work as you can see on the build here https://bafybeie56b7gx4eod6bpbrbwyyvejq7hqtd5apivp5qi2g3zuwdrizoxhy.on.fleek.co/#/delegate/balancer.eth

**Comment by gbonumore** [comment]
Yes this is still WIP thanks for pointing it out

**Comment by gbonumore** [comment]
@samuveth moved back for testing purposes but the last build shows it is working now:
https://bafybeifksq5zkqqjdgfdngqvhbkg2fgjzstka5heryopr4a7ohor4gpjde.on.fleek.co/#/delegate/ybaby.eth

**Comment by samuveth** [comment]
@gbonumore does the balance have no delegates or why doesn't it work there?

**Comment by gbonumore** [comment]
I'll take a look into it

File: src/views/Delegate.vue [hunk 1]
```diff
@@ -198,6 +255,28 @@ onMounted(async () => {
...
+            class="px-4 py-3 border-t flex"
+          >
+            <User
+              :profile="profiles[delegate.delegate]"
+              :address="delegate.delegate"
+              :space="{ network: web3.network.key }"
+              class="column"
+            />
+            <div class="flex-auto column text-right link-color">
+              {{ parseFloat(delegate.score).toFixed(2) }} {{ space.symbol }}
```
Comment (samuveth) [inline src/views/Delegate.vue:276]: We should format this number like we do in Votes with `_n`

File: src/views/Delegate.vue [hunk 2]
```diff
@@ -198,6 +255,28 @@ onMounted(async () => {
             />
           </div>
         </Block>
+        <Block
```
Comment (samuveth) [inline src/views/Delegate.vue:258]: This block should show as loading, and once it's loaded it should only show if `delegatesWithScore.length > 0`

Comment (samuveth) [reply src/views/Delegate.vue:260]: @gbonumore you need to add :loading="delegatesLoading" to the block.

**Comment by gbonumore** [comment]
@samuveth thanks for updating this! I was just having a look at this :)

**Comment by samuveth** [comment]
@gbonumore So because of another PR that I just merged there are some bigger issues that have come up on this PR. I will do some refactoring on my side tomorrow that will make the new space object more easily available for you here.

**Comment by samuveth** [comment]
New branch https://github.com/snapshot-labs/snapshot/pull/688

**Pull request closed without merging**